add Crypto.md file decribes how to port and run crypto tests

netanelgonen opened the pull request:
add Crypto.md file describes how to port and run the crypto tests in mbed-os

Commit:
add Crypto.md

Patater commented:
CC @MarceloSalazar @iriark01

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+* Test for entropy injection feature
+
+In order to run these tests make sure the following configuration is set in your target:
+1.  "extra_labels" contain the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 
```
```diff
-contain
+contains
```

iriark01 commented:
Too late for 5.11, as agreed with Aaron. 

Also needs product approval.

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+1.  "extra_labels" contain the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 
+2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)
+3. "MBEDTLS_ENTROPY_NV_SEED" macro is enabled in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7673)
+4. "MBEDTLS_PLATFORM_NV_SEED_READ_MACRO" macro is set to mbed_default_seed_read" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)
```
Unbalanced `"` here, around `mbed_default_seed_read`.

Patater replied:
```diff
-in wanted
+is wanted
```

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+In order to run these tests make sure the following configuration is set in your target:
+1.  "extra_labels" contain the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 
+2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)
+3. "MBEDTLS_ENTROPY_NV_SEED" macro is enabled in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7673)
```
```diff
-in wanted
+is wanted
```

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)
+3. "MBEDTLS_ENTROPY_NV_SEED" macro is enabled in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7673)
+4. "MBEDTLS_PLATFORM_NV_SEED_READ_MACRO" macro is set to mbed_default_seed_read" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)
+5. "MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO" macro is set to mbed_default_seed_write" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)
```
```diff
-in wanted
+is wanted
```

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+5. "MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO" macro is set to mbed_default_seed_write" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)
+
+### Compile and run
+in order to compile and run the PSA-Crypto Test run the following command:
```
```diff
- in order
+ In order
```

Patater commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+<h2 id="crypto-port">PSA CRYPTO</h2>
+
+For information about PSA-Crypto, please refer to [the PSA-Crypto repository](https://github.com/ARMmbed/mbedtls-psa).
```
Link to `https://github.com/ARMmbed/mbed-crypto` instead.

Also, use "Mbed Crypto" instead of "PSA Crypto" throughout this document.

Commit:
spelling fixes and replace psa-crypto to mbed-crypto

MarceloSalazar commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+
+In order to run these tests make sure the following configuration is set in your target:
+1.  "extra_labels" contains the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 
+2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)
```
Please replace 'Sequana' with 'Future Sequana', here and the whole doc.

netanelgonen replied:
👍

MarceloSalazar reviewed (commented):
Minor tweaks required, but overall looks good.

MarceloSalazar commented:
@Patater thanks for raising this. It looks good.
@iriark01 we identified the need for this doc during OOB and agreed with @AnotherButler to get this in asap.

Commit:
replace Sequana to Future Sequana

iriark01 commented:
As I said, I discussed this with Aaron and it's too late for 5.11

iriark01 commented:
@Patater I understand that the testing bit is what PE asked for, but on its own it's a little context free. As a user, I don't know:

1. Am I supposed to be porting something here, or just testing something that already exists in Mbed OS?
1. When do I run these tests? Why? What's the result I'm looking for?
1. How is this related to the other crypto things we have in porting?
1. Why are we pointing to the repo instead of a specific doc? Looking at the 5.11 OOB, we have specific docs about Crypto.

iriark01 commented:
@Patater ping. Any chance of giving me the rest of the info?

avolinski commented:
Hi @iriark01 ,

Let me try to answer your questions

1. 
Good question, it's a little of both. 
The idea is to have any new added to mbedOS target requiring support for entropy injection to enjoy such functionality and be tested.
Also existing targets that already require such feature can also be tested.

2. 
The tests are running by default only on specifically supported boards.
And can be run manually by user, by specifying additional (as  described) compiler flags from command line to run on a custom board of user choice.
The result is as acceptable in all mbedOS tests, the tests should PASS (FAILED or SKIPPED are 2 other options)

3.
Not sure what other things you have regarding crypto in porting.
But it does relate in terms of adding capability of injecting initial entropy for boards without native HW TRNG support

4.
We can make this PR anywhere you like, if that's what you mean.
If you ask regarding the url links inside the document. They point to a file inside mbedOS.
They do not  point to a document, they need to point to content of a json config file

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,21 @@
+
+In order to run these tests make sure the following configuration is set in your target:
+1.  "extra_labels" contains the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Future Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 
+2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Future Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)
```
@netanelgonen  should we go with exact label name as it is written in json file.
so it would be easier to lookup for the customer?
FUTURE_SEQUANA

netanelgonen replied:
actually, the target name is FEATURE_SEQUANA_PSA, for the lookup the user can use the link. he doesn't need to look for it by himself... 
I don't care about changing it but, I think this is more readable.

Commit:
Edit and add content to crypto.md

Add content from comments, and edit document, mostly for active voice, agreement and formatting.

AnotherButler commented:
@avolinski Please review my changes to make sure I accurately captured the content from your comment. Also, how does this relate to [this page](https://os.mbed.com/docs/mbed-os/v5.11/porting/hardware-accelerated-crypto.html)?

netanelgonen commented:
@avolinski LGTM, do you approve?

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,30 @@
+<h2 id="crypto-port">Mbed Crypto</h2>
+
+Mbed OS provides Mbed Crypto for targets that require support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).
```
it is more of "Mbed Crypto provides capability of entropy injection for target platforms"

netanelgonen replied:
actually, I think that mbed-os is the right way to go, as the user takes mbed-os not mbed-crypto, and may not fully understand what you mean in mbed-crypto. please notice the document refer to tests in mbed-os and not mbed-crypto

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,30 @@
+
+Both targets newly ported to Mbed OS and existing targets that use entropy injection require passing tests to validate Mbed Crypto functionality.
+
+We run tests by default on some supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```
```suggestion
We run tests by default on several supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,30 @@
+<h2 id="crypto-port">Mbed Crypto</h2>
+
+Mbed OS provides Mbed Crypto for targets that require support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).
```
```suggestion
Mbed Crypto library as part of Mbed OS provides support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).
```

netanelgonen replied:
the same comment as above, those tests are part of mbed-os and not mbed-crypto. I think mbed-os is correct.

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -0,0 +1,30 @@
+
+Both targets newly ported to Mbed OS and existing targets that use entropy injection require passing tests to validate Mbed Crypto functionality.
+
+We run tests by default on some supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```
```suggestion
We run tests by default on a list of supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```

netanelgonen replied:
maybe keep the same as above, 'several'? it will be better to be consistent.

alekshex commented on docs/porting/psa/crypto.md:
```diff
@@ -1,21 +1,30 @@
+- A test for initialization of the Mbed Crypto module.
+- A test for entropy injection feature.
+
+To run these tests, make sure the following configuration is set in your target:
```
To run these tests, make sure your target configuration is set as follows:

netanelgonen replied:
there is no difference between those two, do we need an additional commit for this?

Commit:
Update docs/porting/psa/crypto.md

replace some with several

Co-Authored-By: netanelgonen <netanel.gonen@arm.com>

avolinski commented:
> @avolinski Please review my changes to make sure I accurately captured the content from your comment. Also, how does this relate to [this page](https://os.mbed.com/docs/mbed-os/v5.11/porting/hardware-accelerated-crypto.html)?

Hi @AnotherButler,
Thank you for your help!
I've made few small comments.
as to your question - 
there is no relation. these features are orthogonal.

AnotherButler commented:
@avolinski and @netanelgonen Are all your comments agreed upon and resolved?

netanelgonen commented:
@avolinski did you see my comments?

avolinski commented:
Hi @AnotherButler ,
My comments are only small suggestions, please pick the ones you think are most suitable.
Leaving it as it is, is also fine.

Commit:
Update crypto.md

Clarify content from comments.
